provider/dynamodb.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+
+	"context"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+	"github.com/cenk/backoff"
+	"github.com/containous/traefik/job"
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/safe"
+	"github.com/containous/traefik/types"
+	"time"
+)
+
+var _ Provider = (*DynamoDB)(nil)
+
+// DynamoDB holds configuration for DynamoDB provider.
+type DynamoDB struct {
+	BaseProvider `mapstructure:",squash"`
+
+	AccessKeyID     string `description:"The AWS credentials access key to use for making requests"`
+	RefreshSeconds  int    `description:"Polling interval (in seconds)"`
+	Region          string `description:"The AWS region to use for requests"`
+	SecretAccessKey string `description:"The AWS credentals secret key to use for making requests"`
+	TableName       string `description:"The AWS dynamodb table that stores configuration for traefik"`
+}
+
+type dynamoClient struct {
+	db dynamodbiface.DynamoDBAPI
+}
+
+// createClient configures aws credentials and creates a dynamoClient
+func (provider *DynamoDB) createClient() *dynamoClient {
+	log.Infof("Creating DynamoDB client...")
+	sess := session.New()
+	if provider.Region == "" {
+		log.Debug("No Region provided. Using us-east-1...")
+		provider.Region = "us-east-1"
+	}
+	cfg := &aws.Config{
+		Region: &provider.Region,
+		Credentials: credentials.NewChainCredentials(
+			[]credentials.Provider{
+				&credentials.StaticProvider{
+					Value: credentials.Value{
+						AccessKeyID:     provider.AccessKeyID,
+						SecretAccessKey: provider.SecretAccessKey,
+					},
+				},
+				&credentials.EnvProvider{},
+				&credentials.SharedCredentialsProvider{},
+				defaults.RemoteCredProvider(*(defaults.Config()), defaults.Handlers()),
+			}),
+	}
+
+	return &dynamoClient{
+		dynamodb.New(sess, cfg),
+	}
+}
+
+// scanTable scans the given table and returns slice of all items in the table
+func (provider *DynamoDB) scanTable(client *dynamoClient) ([]map[string]*dynamodb.AttributeValue, error) {
+	log.Debugf("Scanning table: %s ...", provider.TableName)
+	params := &dynamodb.ScanInput{
+		TableName: aws.String(provider.TableName),
+	}
+	items := make([]map[string]*dynamodb.AttributeValue, 0)
+	err := client.db.ScanPages(params,
+		func(page *dynamodb.ScanOutput, lastPage bool) bool {
+			items = append(items, page.Items...)
+			return !lastPage
+		})
+	if err != nil {
+		log.Errorf("Failed to scan table %s", provider.TableName)
+		return nil, err
+	}
+	log.Debugf("Successfully scanned %s table", provider.TableName)
+	return items, nil
+}
+
+// loadDynamoConfig retrieves items from dynamodb and converts them into Backends and Frontends in a Configuration
+func (provider *DynamoDB) loadDynamoConfig(client *dynamoClient) (*types.Configuration, error) {
+	items, err := provider.scanTable(client)
+	if err != nil {
+		return nil, err
```

Comment:
I added a flag in the mock ScanPages to return an error and trigger this code.
https://github.com/containous/traefik/pull/1158/files#diff-57d4449dc0fc79752ce983c308204295R45